src/Sulu/Bundle/MediaBundle/Media/Storage/StorageInterface.php

```diff
@@ -22,29 +22,37 @@
      *
      * @param $tempPath
      * @param $fileName
-     * @param $version
+     * @param $preferredStorageOptions
      *
      * @return mixed
      */
-    public function save($tempPath, $fileName, $version);
+    public function save($tempPath, $fileName, $preferredStorageOptions = null);
 
     /**
-     * Give back the path to the document.
+     * Will return a file resource.
      *
-     * @param $fileName
-     * @param $version
-     * @param $storageOption
+     * @param $storageOptions
      *
-     * @return mixed
+     * @return resource
      */
-    public function load($fileName, $version, $storageOption);
+    public function load($storageOptions);
```

Comment:
IMHO `save($tempPath, $preferedStorageOptions = null)`. I don't see the advantage of the second version.

But as said above, for the current implementations the file name seems mandatory. I don't see how the storage manager (or the user of the storage manager) should determine if it should pass the file name or the id (in future cases)?

:grin: